### ISSUE TITLE
Allow OP to mark comment as spoiler

### ISSUE THREAD

**Issue by local-minimum** [ludumdare/ludumdare#1801]
This is particularly for games-posts, it happens that people possibly inadvertently spoils the entire game. In the example that triggered this issue (https://ldjam.com/events/ludum-dare/44/utsvulten) the spoiler is in the form of an image.

## Problems
1. The spoiler mark down, doesn't work with hiding images.
2. The chance that the one can manage to get a hold of the one who spoiled to fix the issue is extremely slight.
3. Spoiler markdown (`!>`) isn't documented anywhere. https://ldjam.com/markdown should explain it since it is not part of standard github markdown.

## Solution
The author of the post may flag a comment as spoiler, this will add an overlay over the entire post that needs clicking to view.
After author marked a comment as spolier, it could look something like this:
![ld_spoiler](https://user-images.githubusercontent.com/8041100/56895663-05644800-6a8a-11e9-90a7-d626fb4db0dd.png)
(Keep space of comment, but have minimum space needed for spoils text)

The author could have a 🏴 icon next to the heart icons on all comments that would popup a dialogue asking if the author is sure they want to flag the comment as spoiling.

My idea is that spoiling is a property of the comment sent with the comment. No filtering is done by the API on getting comments. This is only a display property client-side. The post request from the author to set or unset the spoiler flag is of course authenticated against being the author of the original post.

## Misuse
Someone might find it fun to flag non-spoiling comments as spoiling, but the data is still sent to the user and visible after clicking.

## Further thoughts
This could of course be extended to not only flag something as spoiling, but also adult, abusive and so on, but those might be more sensitive areas whereas spoiling should be rather benign.

**Comment by mikekasprzak** [comment]
Good idea. :+1:

**Comment by local-minimum** [comment]
@mikekasprzak you implied before that after this LD you'd have more time again so we could perhaps deal with the old PRs and then work on this one? If probably need help with the backend API changes that would be needed.

**Comment by mikekasprzak** [comment]
Yeah I have some personal stuff to take care of over the next couple
weekends, but yes I'd like to start getting some momentum on the website
going again. I'll have more to say soon, but I'd like to do a bit of a
postmortem of what does and doesn't work well about the website, the
codebase, and so on, and get your and everyone else feedback. This past
year I've done a lot of work with scalable servers, and I want to do a
better job here, but that's topic all its own.

**Comment by local-minimum** [comment]
For pt3 above, I guess it's an update needed in the database text for that special page or? I had to teach another participant today the `!>`.

For getting started again on site-dev, perhaps it be good do a discord voice meeting or similar to sort out what we all need and want.
